GLOSSARY.md

```diff
@@ -0,0 +1,27 @@
+# Glossary
+
+This document gathers the canonical nomenclature of the Lightning concepts used by the LDK contributors. Note, they may diverge from the BOLT specs or the ones used by the wider Lightning community, the test of time has revealed that it can be hard to agree on terms employed. The following terms are employed with a best effort across the codebase aiming to reduce both code reviewers and library users confusion.
+
+### Holder/Counterparty
+
+Inspired by financial cryptography, the holder is the entity beneficiary of a right. In LDK context, it designates the entity who is served by the implementation in its channel operations. The counterparty is the entity participating in the channel operation as a peer, but of whom the interests are not of concern by the implementation. Those roles are fixed for the lifetime of a channel and doesn't change in fonction of the state (message, blocks, ...) processed.
```

Comment:
s/fonction/function/

also, not sure it makes sense to say that the roles are fixed, since by definition they must be fixed.  since the holder is the entity that the implementation is concerned about, it couldn't possibly change because the implementation can't start being concerned about the other party.

Reply:
The use of `designates` was a bit confusing for me as it's ambiguous if its defining holder or indicating that the holder is selecting an entity.  Suggest simplifying to `In LDK context, the holder is the entity`

Reply:
Done.